frame/fee-market/src/lib.rs

```diff
@@ -184,10 +181,11 @@ pub mod pallet {
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_finalize(_: BlockNumberFor<T>) {
 			// Clean the order's storage when the rewards has been paid off
-			for (lane_id, message_nonce) in <ConfirmedMessagesThisBlock<T>>::get() {
-				<Orders<T>>::remove((lane_id, message_nonce));
+			for ((lane_id, message_nonce), order) in <Orders<T>>::iter() {
+				if order.confirm_time.is_some() {
```

Comment:
Add documents, if confirm_time is some, meaning this order has already been processed.